src/kv/store.h

```diff
@@ -1058,6 +1042,26 @@ namespace kv
       }
     }
 
+    bool must_force_ledger_chunk(Version version) override
+    {
+      std::lock_guard<std::mutex> vguard(version_lock);
+      return must_force_ledger_chunk_unsafe(version);
+    }
+
+    bool must_force_ledger_chunk_unsafe(Version version) override
+    {
```

Comment:
I think this needs a comment to specify that `version` has to be of a committable entry (i.e. signature).

Reply:
Yes, I agree. Is there a way to automatically check that?

Reply:
Comment added, but I think it would be nice if we could check that automatically. In theory we could save the txid of the reserved TX when we created it, but it feels like that might trigger a bunch of other work to make sure there aren't multiple of them in flight, etc.

Reply:
Alternatively, the `force_ledger_chunk` flag could be passed in the constructor of the `ReservedTx` when we call `store->create_reserved_tx(tx_id)` since we could ask the snapshotter early whether a the transaction will require the creation of a new ledger chunk.

Reply:
Hmm. Could that potentially create problems when the transaction doesn't get committed?